deployment/cdk/opensearch-service-migration/lib/service-stacks/opensearch-container-stack.ts

```diff
@@ -0,0 +1,69 @@
+import {StackPropsExt} from "../stack-composer";
+import {IVpc, SecurityGroup} from "aws-cdk-lib/aws-ec2";
+import {PortMapping, Protocol, Ulimit, UlimitName} from "aws-cdk-lib/aws-ecs";
+import {Construct} from "constructs";
+import {MigrationServiceCore} from "./migration-service-core";
+import {StringParameter} from "aws-cdk-lib/aws-ssm";
+import {ServiceConnectService} from "aws-cdk-lib/aws-ecs/lib/base/base-service";
+
+export interface OpenSearchContainerProps extends StackPropsExt {
+    readonly vpc: IVpc
+}
+
+/**
+ * This is a testing stack which deploys a SINGLE node OpenSearch cluster as the target cluster for this solution with
+ * no persistent storage. As such, it should NOT be used for production use cases.
+ */
+export class OpenSearchContainerStack extends MigrationServiceCore {
+
+    constructor(scope: Construct, id: string, props: OpenSearchContainerProps) {
+        super(scope, id, props)
+        let securityGroups = [
+            SecurityGroup.fromSecurityGroupId(this, "serviceConnectSG", StringParameter.valueForStringParameter(this, `/migration/${props.stage}/${props.defaultDeployId}/serviceConnectSecurityGroupId`)),
+        ]
+
+        const servicePort: PortMapping = {
+            name: "opensearch-connect",
+            hostPort: 9200,
+            containerPort: 9200,
+            protocol: Protocol.TCP
+        }
+        const serviceConnectService: ServiceConnectService = {
+            portMappingName: "opensearch-connect",
+            dnsName: "opensearch",
+            port: 9200
+        }
+        const ulimits: Ulimit[] = [
+            {
+                name: UlimitName.MEMLOCK,
+                softLimit: -1,
+                hardLimit: -1
+            },
+            {
+                name: UlimitName.NOFILE,
+                softLimit: 65536,
+                hardLimit: 65536
```

Comment:
will these limits be supported on the container that OS is running on?  What's the behavior if the process breaches these bounds what happens?

Reply:
Fargate supports overriding default ulimits: https://stackoverflow.com/questions/65289458/ulimits-on-aws-ecs-fargate, which is what I'm utilizing here with defaults that opensearch uses on their website. I'm not sure though on what happens if we were to try and say breach the NOFILE limit

Reply:
I guess the worst case scenario is that the process dies because it exhausts one of those limits, which it would have probably done when it bumped into the limit anyway.